===================================================================
RCS file: /GraphicsMagick/GraphicsMagick/coders/xwd.c,v
retrieving revision 1.88
retrieving revision 1.88.2.1
diff -u -p -r1.88 -r1.88.2.1
--- GraphicsMagick/coders/xwd.c	2004/02/17 03:41:53	1.88
+++ GraphicsMagick/coders/xwd.c	2005/05/16 18:51:04	1.88.2.1
@@ -325,21 +325,21 @@ static Image *ReadXWDImage(const ImageIn
         */
         red_mask=ximage->red_mask;
         red_shift=0;
-        while ((red_mask & 0x01) == 0)
+        while ((red_mask != 0) && ((red_mask & 0x01) == 0))
         {
           red_mask>>=1;
           red_shift++;
         }
         green_mask=ximage->green_mask;
         green_shift=0;
-        while ((green_mask & 0x01) == 0)
+        while ((green_mask != 0) && ((green_mask & 0x01) == 0))
         {
           green_mask>>=1;
           green_shift++;
         }
         blue_mask=ximage->blue_mask;
         blue_shift=0;
-        while ((blue_mask & 0x01) == 0)
+        while ((blue_mask != 0) && ((blue_mask & 0x01) == 0))
         {
           blue_mask>>=1;
           blue_shift++;
